src/script/common/c_content.cpp

```diff
@@ -1959,8 +1959,18 @@ void push_objectRef(lua_State *L, const u16 id)
 
 void read_hud_element(lua_State *L, HudElement *elem)
 {
-	elem->type = (HudElementType)getenumfield(L, 2, "hud_elem_type",
-									es_HudElementType, HUD_ELEM_TEXT);
+	// Handle deprecated hud_elem_type
+	std::string type_string;
+	if (getstringfield(L, 2, "hud_elem_type", type_string)) {
+		log_deprecated(L, "Deprecated \"hud_elem_type\" field, use \"type\" instead.");
+		int type_enum;
+		if (string_to_enum(es_HudElementType, type_enum, type_string))
+			elem->type = (HudElementType)type_enum;
+		else
+			elem->type = HUD_ELEM_TEXT;
+	} else {
+		elem->type = (HudElementType)getenumfield(L, 2, "type", es_HudElementType, HUD_ELEM_TEXT);
```

Comment:
What if both are specified? Suggestion: after L1961:  `elem->type = HUD_ELEM_TEXT;` then use `elem->type` as default here.

Reply:
I have changed it so that in case it is ambiguous the new "type" filed will be used. (You also get a custom error in this case.)
If neither "type" nor "hud_elem_type" the type is HUD_ELEM_TEXT like it was before.